apps/system/test/marionette/pinning_the_web_test.js

```diff
@@ -103,4 +104,58 @@ marionette('Pinning the Web', function() {
     });
   });
 
+  suite('Unpin site', function() {
+    var url, lastIconId;
+
+    function lastIconMatches(id) {
+      system.tapHome();
+      client.switchToFrame(system.getHomescreenIframe());
+      client.waitFor(function() {
+        var ids = home.getIconIdentifiers();
+        return id == ids[ids.length-1];
+      });
+    }
+
+    setup(function() {
+      system.tapHome();
+      client.switchToFrame(system.getHomescreenIframe());
+      var ids = home.getIconIdentifiers();
+      lastIconId = ids[ids.length-1];
+
+      client.switchToFrame();
+      url = server.url('sample.html');
+      pinTheWeb.openAndPinSite(url);
+      lastIconMatches(url);
+    });
+
+    test('Unpin site', function() {
+      // Check that the pin door hanger appears
+      system.tapHome();
+      rocketbar.homescreenFocus();
+      rocketbar.enterText(url, true);
+      system.gotoBrowser(url);
+      client.switchToFrame();
+      client.scope({ searchTimeout: 100 }).waitFor(function() {
+        client.switchToFrame();
+        try {
+          system.siteIcon.tap();
+        } catch (e) {
+          return false;
+        }
+        return true;
+      });
+      actions.wait(1);
```

Comment:
Can we wait for the doorhanger displayed, and assert that the action of the button is 'unpin'?